src/main/java/Duke.java

```diff
@@ -1,10 +1,236 @@
+                + " |     \\   |  |__/ __ \\_\\___ \\|   Y  \\\n"
+                + " \\___  /   |____(____  /____  >___|  /\n"
+                + "     \\/              \\/     \\/     \\/\n";
+        System.out.println("Hello from\n"+ logo);
+        String lineDivider = "____________________________________________________________\n";
+        System.out.println("\t" + lineDivider
+                + "\t Hello! I'm Flash\n"
+                + "\t What can I do for you?\n"
+                + "\t" + lineDivider);
+        return lineDivider;
+    }
+
+    /**
+     * Prints out a list of the tasks saved from the user inputs.
+     */
+    public static void printList() {
+        int numberOrder = 1;
+
+        System.out.println("\t Here are the tasks in your list:");
+
+        for (int i = 0; i < indexTask; i++) {
+            System.out.println("\t " + numberOrder + ".["
+                    + assignments[i].getStatusOfTypeTask() + "]["
+                    + assignments[i].getStatusOfDone() + "] "
+                    + assignments[i].displayTypeTaskDetails());
+            numberOrder++;
+        }
+    }
+
+    /**
+     * Prints a message in CLI to inform the user if the task has been marked or not.
+     *
+     * @param markIndex index of which task to mark in splitUserInput[1].
+     * @param isMark boolean true or false to check if the task has been marked.
+     */
+    public static void printMarkOrUnmarkTask(int markIndex, boolean isMark) {
+        if (isMark) {
+            System.out.println("\t Awesome! I've marked this task as done:");
+        } else {
+            System.out.println("\t Awesome! I've marked this task as not done yet:");
+        }
+        System.out.println("\t\t [" + assignments[markIndex].getStatusOfTypeTask()
+                + "]" + "[" + assignments[markIndex].getStatusOfDone()
+                + "] " + assignments[markIndex].description);
+    }
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+
+        String lineDivider = printWelcomeMessage();
+        String userInput;
+        Scanner in = new Scanner(System.in);
+        boolean isBye;
+
+        do {
+            //Enable user to enter text
+            userInput = in.nextLine();
+            System.out.print("\t" + lineDivider);
+            String[] splitUserInputs = userInput.split(" ", 2);
+            isBye = splitUserInputs[0].equals("bye");
+
+            boolean isUserInputData = !splitUserInputs[0].equals("bye")
+                    && !splitUserInputs[0].equals("list")
+                    && !splitUserInputs[0].equals("mark")
+                    && !splitUserInputs[0].equals("unmark");
```

Comment:
Avoid complicated expressions. Perhaps you can consider splitting the logic up into multiple statements and abstract similar methods (SLAP) to avoid repeating the code. E.g. a function `isUserInputValid(string userInput)` can be used to check for valid user input.